### PR TITLE
Feature/64 as doctor see patient info

### PR DESCRIPTION
**Pull request by TREN-D**

### CODE REVIEW

**Review by SergiyKozachok — APPROVED** [review]

**Review by eugent482 — APPROVED** [review]

File: InternetHospital.BusinessLogic/Services/AppointmentService.cs [hunk 1]
```diff
@@ -52,6 +52,71 @@ public IEnumerable<AppointmentModel> GetMyAppointments(int doctorId)
...
+        /// </returns>
+        public PageModel<IEnumerable<AppointmentModel>> GetMyAppointments(AppointmentHistoryParameters parameters, int doctorId)
+        {
+            DeleteUncommittedAppointments(doctorId);
+            var appointments = _context.Appointments
+               .OrderByDescending(a => a.StartTime)
+               .Include(a => a.IllnessHistory)
+               .Include(a => a.User)
+               .Include(a => a.Status)
+               .Where(a => a.DoctorId == doctorId);
```
Comment (voznovych) [inline InternetHospital.BusinessLogic/Services/AppointmentService.cs:70]: It's better to include after filtering by where and order in the end.

File: InternetHospital.BusinessLogic/Services/DoctorService.cs [hunk 2]
```diff
@@ -240,61 +240,86 @@ public async Task<bool> UpdateDoctorInfo(DoctorProfileModel doctorModel, int use
...
-                .Include(a => a.User)
-                .Include(a => a.Status)
-                .Where(a => a.DoctorId == doctorId);
-
-            if (!string.IsNullOrEmpty(parameters.SearchByName))
+            var appointment = _context
+                                .Appointments
+                                .Where(a => a.UserId == userId
+                                        && a.DoctorId == doctorId
+                                        && a.IsAllowPatientInfo == true
```
Comment (voznovych) [inline InternetHospital.BusinessLogic/Services/DoctorService.cs:249]: Why not "a.IsAllowPatientInfo"?

File: InternetHospital.BusinessLogic/Services/DoctorService.cs [hunk 3]
```diff
@@ -240,61 +240,86 @@ public async Task<bool> UpdateDoctorInfo(DoctorProfileModel doctorModel, int use
...
+            return patientProfile;
+        }
+
+        public PageModel<IEnumerable<IllnessHistoryModel>> GetPatientIllnessHistory(IllnessHistorySearchModel queryParameters, int doctorId)
+        {
+            var appointment = _context
+                    .Appointments
+                    .Where(a => a.UserId == queryParameters.UserId
+                            && a.DoctorId == doctorId
+                            && a.IsAllowPatientInfo == true
```
Comment (voznovych) [inline InternetHospital.BusinessLogic/Services/DoctorService.cs:270]: Same.

File: InternetHospital.BusinessLogic/Services/DoctorService.cs [hunk 4]
```diff
@@ -240,61 +240,86 @@ public async Task<bool> UpdateDoctorInfo(DoctorProfileModel doctorModel, int use
...
+                            && a.StatusId == (int)AppointmentStatuses.RESERVED_STATUS)
+                    .Include(a => a.User)
+                        .ThenInclude(u => u.IllnessHistories)
+                    .SingleOrDefault();
+
+            if (appointment == null)
             {
-                appointments = appointments
-                    .Where(a => a.StartTime >= parameters.From.Value);
+                return null;
```
Comment (voznovych) [inline InternetHospital.BusinessLogic/Services/DoctorService.cs:278]: Return empty object.

File: InternetHospital.BusinessLogic/Services/DoctorService.cs [hunk 5]
```diff
@@ -240,61 +240,86 @@ public async Task<bool> UpdateDoctorInfo(DoctorProfileModel doctorModel, int use
...
+            var historiesResult = PaginationHelper(illnessHistories, queryParameters.PageCount, queryParameters.Page).ToList();
 
-            return new PageModel<IEnumerable<PreviousAppointmentsModel>>()
-            { EntityAmount = appointmentsAmount, Entities = appointmentsResult };
+            return new PageModel<IEnumerable<IllnessHistoryModel>>()
+            { EntityAmount = historyAmount, Entities = historiesResult };
         }
 
-        public IEnumerable<string> GetAppointmentStatuses()
+        private IQueryable<IllnessHistoryModel> PaginationHelper(IQueryable<IllnessHistory> histories, int pageCount, int page)
```
Comment (voznovych) [inline InternetHospital.BusinessLogic/Services/DoctorService.cs:304]: There is Helper for pagination. This method is mapper.

File: InternetHospital.WebApi/Controllers/AppointmentsController.cs [hunk 6]
```diff
@@ -30,7 +30,35 @@ public IActionResult GetAppointments()
...
+        {
+            if (!int.TryParse(User.Identity.Name, out int doctorId))
+            {
+                return BadRequest(new { message = "Wrong claims" });
+            }
+
+            var result = _appointmentService.GetMyAppointments(parameters, doctorId);
+
+            return Ok(
+                new
```
Comment (voznovych) [inline InternetHospital.WebApi/Controllers/AppointmentsController.cs:45]: Why not result?

File: InternetHospital.WebApi/Controllers/AppointmentsController.cs [hunk 7]
```diff
@@ -30,7 +30,35 @@ public IActionResult GetAppointments()
...
+                }
+            );
+        }
+
+        [Authorize(Policy = "ApprovedDoctors")]
+        [HttpGet("appointmentStatuses")]
+        public IActionResult GetAppointmentStatuses()
+        {
+            var statuses = _appointmentService.GetAppointmentStatuses();
+            return Ok(statuses);
```
Comment (voznovych) [inline InternetHospital.WebApi/Controllers/AppointmentsController.cs:58]: Just Ok(_appointmentService.GetAppointmentStatuses());

File: InternetHospital.WebApi/Controllers/DoctorsController.cs [hunk 8]
```diff
@@ -144,32 +144,38 @@ public IActionResult FillHistory([FromBody] IllnessHistoryModel illnessHistory)
...
-            var statuses = _doctorService.GetAppointmentStatuses();
-            return Ok(statuses);
+            if (!int.TryParse(User.Identity.Name, out int doctorId))
+            {
+                return BadRequest();
+            }
+
+            var illnessHistories = _doctorService.GetPatientIllnessHistory(queryParameters, doctorId);
+
+            return illnessHistories != null ?
```
Comment (voznovych) [inline InternetHospital.WebApi/Controllers/DoctorsController.cs:172]: I don't think that ternary operator is good in this case.

File: InternetHospital.WebApi/Startup.cs [hunk 9]
```diff
@@ -199,9 +199,9 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
                 config.CreateMap<Article, SendingArticleEditingModel>();
                 config.CreateMap<IllnessHistoryModel, IllnessHistory>();
                 config.CreateMap<IllnessHistory, IllnessHistoryModel>();
-                config.CreateMap<Appointment, PreviousAppointmentsModel>()
-                .ForMember(prevAppoint => prevAppoint.UserFullName, opt => opt.MapFrom(o => $"{o.User.FirstName} {o.User.SecondName}"))
-                .ForMember(prevAppoint => prevAppoint.Status, opt => opt.MapFrom(o => o.Status.Name));
+                config.CreateMap<Appointment, AppointmentModel>()
+                .ForMember(appointment => appointment.Status, opt => opt.MapFrom(o => o.Status.Name));
```
Comment (voznovych) [inline InternetHospital.WebApi/Startup.cs:203]: Tab.

File: InternetHospital.WebApi/appsettings.json [hunk 10]
```diff
@@ -1,7 +1,7 @@
 {
   "ConnectionStrings": {
     "AzureConnection": "Server=tcp:itacad.database.windows.net,1433;Initial Catalog=HospitalDev;Persist Security Info=False;User ID=system_admin;Password=$oft$erve1234;MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30",
-    "LocaleMsSql": ""
+    "LocaleMsSql": "Data Source=(localdb)\\ProjectsV13;Initial Catalog=HospitalDev;Integrated Security=True;Connect Timeout=30;Encrypt=False;TrustServerCertificate=False;ApplicationIntent=ReadWrite;MultiSubnetFailover=False"
```
Comment (OleksandrAntoniuk) [inline InternetHospital.WebApi/appsettings.json:4]: LocalDbConnection

**Review by OleksandrAntoniuk — APPROVED** [review]

**Review by voznovych — APPROVED** [review]

**Pull request merged**